src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/gps/GPSMarkerTool.java

```diff
@@ -36,7 +36,11 @@ public ItemUseHandler getItemHandler() {
 
             if (e.getClickedBlock().isPresent()) {
                 Block b = e.getClickedBlock().get().getRelative(e.getClickedFace());
-                Slimefun.getGPSNetwork().createWaypoint(e.getPlayer(), b.getLocation());
+                Location l = b.getLocation();
+                Location locationWithYaw = new Location(
+                    l.getWorld(), l.getX(), l.getY(), l.getZ(), e.getPlayer().getLocation().getYaw(), 0.0f
+                );
```

Comment:
Rather than making a new location aren't you able to call setYaw on l? Block locations don't change the block so I think that's all that's required here